### PR TITLE
Vital signs, diagnosis and saving encounter files fixes

### PR DESCRIPTION
**Pull request by mateuszkrasucki**
FIX no error returned from on file upload failed
ADD method for updating waitlist history items
UPD vitalSigns migration to migrate history
FIX diagnosis and vitalSigns migration
FIX waitlist vitalSigns timestamp
ADD saving vitalSigns timestamp to encounter

### CODE REVIEW

File: frontend/local/src/containers/waitlist/detail/add-data.js [hunk 1]
```diff
@@ -80,7 +81,7 @@ class MedicalData extends React.Component {
                     10000,
                 2
             )
-            vitalSigns.bmi.timestamp = (moment(vitalSigns.height.timestamp, "X").isAfter(moment(vitalSigns.weight.timestamp, "X")) ? moment(vitalSigns.height.timestamp, "X") : moment(vitalSigns.weight.timestamp, "X")).format("X")
+            vitalSigns.bmi.timestamp = (moment(vitalSigns.height.timestamp).isAfter(moment(vitalSigns.weight.timestamp)) ? moment(vitalSigns.height.timestamp) : moment(vitalSigns.weight.timestamp)).format()
```
Comment (dominikznidar) [inline frontend/local/src/containers/waitlist/detail/add-data.js]: `moment` also supports `min` and `max` functions

File: frontend/local/src/modules/ehr.js [hunk 2]
```diff
@@ -99,6 +99,10 @@ const specToDocument = (specs, data, formData, ehrPrefix) => {
                 acc.push(assignBoolean(ehrPrefix + spec.ehrPath, spec.formPath))
                 break
 
+            case "dateTime":
+                acc.push(assignBoolean(ehrPrefix + spec.ehrPath, spec.formPath))
```
Comment (dominikznidar) [inline frontend/local/src/modules/ehr.js]: `assign*` function should be changed

File: frontend/local/src/modules/ehr/encounter.js [hunk 3]
```diff
@@ -94,6 +94,12 @@ export default dispatch =>
             unit: "kg",
             formPath: "vitalSigns.weight.value"
         },
+        {
+            type: "dateTime",
+            ehrPath:
+                "/content[openEHR-EHR-COMPOSITION.encounter.v1]/context/other_context/items[openEHR-EHR-OBSERVATION.body_weight.v2]/context/start_time",
+            formPath: "vitalSigns.weight.timestamp"
+        },
```
Comment (dominikznidar) [inline frontend/local/src/modules/ehr/encounter.js]: I made a few mistakes with these data points.
Quantity should be stored as 
```
/content[openEHR-EHR-COMPOSITION.encounter.v1]/context/other_context/items[openEHR-EHR-OBSERVATION.body_weight.v2]/data[at0002]/events[at0003]:0/data[at0001]/items[at0004]
```
and the time as
```
/content[openEHR-EHR-COMPOSITION.encounter.v1]/context/other_context/items[openEHR-EHR-OBSERVATION.body_weight.v2]/data[at0002]/events[at0003]:0/time
```

File: frontend/local/src/modules/ehr/encounter.js [hunk 4]
```diff
@@ -103,6 +109,12 @@ export default dispatch =>
             unit: "cm",
             formPath: "vitalSigns.height.value"
         },
+        {
+            type: "dateTime",
+            ehrPath:
+                "/content[openEHR-EHR-COMPOSITION.encounter.v1]/context/other_context/items[openEHR-EHR-OBSERVATION.height.v2]/context/start_time",
+            formPath: "vitalSigns.height.timestamp"
+        },
```
Comment (dominikznidar) [inline frontend/local/src/modules/ehr/encounter.js]: value should be stored as 
```
/content[openEHR-EHR-COMPOSITION.encounter.v1]/context/other_context/items[openEHR-EHR-OBSERVATION.height.v2]/data[at0001]/events[at0002]:0/data[at0003]/items[at0004]
```
timestamp should be stored as 
```
/content[openEHR-EHR-COMPOSITION.encounter.v1]/context/other_context/items[openEHR-EHR-OBSERVATION.height.v2]/data[at0001]/events[at0002]:0/time
```

File: frontend/local/src/modules/ehr/encounter.js [hunk 5]
```diff
@@ -112,6 +124,12 @@ export default dispatch =>
             unit: "kg/m2",
             formPath: "vitalSigns.bmi.value"
         },
+        {
+            type: "dateTime",
+            ehrPath:
+                "/content[openEHR-EHR-COMPOSITION.encounter.v1]/context/other_context/items[openEHR-EHR-OBSERVATION.body_mass_index.v2]/context/start_time",
+            formPath: "vitalSigns.bmi.timestamp"
+        },
```
Comment (dominikznidar) [inline frontend/local/src/modules/ehr/encounter.js]: value should be stored as
```
/content[openEHR-EHR-COMPOSITION.encounter.v1]/context/other_context/items[openEHR-EHR-OBSERVATION.body_mass_index.v2]/data[at0001]/events[at0002]:0/data[at0003]/items[at0004]
```

timestamp as
```
/content[openEHR-EHR-COMPOSITION.encounter.v1]/context/other_context/items[openEHR-EHR-OBSERVATION.body_mass_index.v2]/data[at0001]/events[at0002]:0/time
```

File: frontend/local/src/modules/ehr/encounter.js [hunk 6]
```diff
@@ -121,6 +139,12 @@ export default dispatch =>
             unit: "Cel",
             formPath: "vitalSigns.temperature.value"
         },
+        {
+            type: "dateTime",
+            ehrPath:
+                "/content[openEHR-EHR-COMPOSITION.encounter.v1]/context/other_context/items[openEHR-EHR-OBSERVATION.temperature.v2]/context/start_time",
+            formPath: "vitalSigns.temperature.timestamp"
+        },
```
Comment (dominikznidar) [inline frontend/local/src/modules/ehr/encounter.js]: value should be stored as:
```
/content[openEHR-EHR-COMPOSITION.encounter.v1]/context/other_context/items[openEHR-EHR-OBSERVATION.body_temperature.v2]/data[at0002]/events[at0003]:0/data[at0001]/items[at0004]
```

timestamp as
```
/content[openEHR-EHR-COMPOSITION.encounter.v1]/context/other_context/items[openEHR-EHR-OBSERVATION.body_temperature.v2]/data[at0002]/events[at0003]:0/time
```

File: frontend/local/src/modules/ehr/encounter.js [hunk 7]
```diff
@@ -139,6 +163,12 @@ export default dispatch =>
             unit: "mm[Hg]",
             formPath: "vitalSigns.pressure.value.diastolic"
         },
+        {
+            type: "dateTime",
+            ehrPath:
+                "/content[openEHR-EHR-COMPOSITION.encounter.v1]/context/other_context/items[openEHR-EHR-OBSERVATION.blood_pressure.v1]/context/start_time",
+            formPath: "vitalSigns.pressure.timestamp"
+        },
```
Comment (dominikznidar) [inline frontend/local/src/modules/ehr/encounter.js]: data should be stored as:
```
/content[openEHR-EHR-COMPOSITION.encounter.v1]/context/other_context/items[openEHR-EHR-OBSERVATION.blood_pressure.v1]/data[at0001]/events[at0006]:0/data[at0003]/items[at0004]
```

timestamp as:
```
/content[openEHR-EHR-COMPOSITION.encounter.v1]/context/other_context/items[openEHR-EHR-OBSERVATION.blood_pressure.v1]/data[at0001]/events[at0006]:0/time
```

File: frontend/local/src/modules/ehr/encounter.js [hunk 8]
```diff
@@ -154,12 +184,24 @@ export default dispatch =>
             unit: "/min",
             formPath: "vitalSigns.heart_rate.value"
         },
+        {
+            type: "dateTime",
+            ehrPath:
+                "/content[openEHR-EHR-COMPOSITION.encounter.v1]/context/other_context/items[openEHR-EHR-OBSERVATION.pulse.v1]/context/start_time",
+            formPath: "vitalSigns.pulse.timestamp"
+        },
```
Comment (dominikznidar) [inline frontend/local/src/modules/ehr/encounter.js]: events requires a the same `:0` suffix as the rest of observations. timestamp should be stored in the same way as the measurements above.

File: frontend/local/src/modules/ehr/encounter.js [hunk 9]
```diff
@@ -154,12 +184,24 @@ export default dispatch =>
...
             type: "value",
             ehrPath:
                 "/content[openEHR-EHR-COMPOSITION.encounter.v1]/context/other_context/items[openEHR-EHR-OBSERVATION.pulse_oximetry.v1]/data[at0001]/events[at0002]/data[at0003]/items[at0006]",
             formPath: "vitalSigns.oxygen_saturation.value"
+        },
+        {
+            type: "dateTime",
+            ehrPath:
+                "/content[openEHR-EHR-COMPOSITION.encounter.v1]/context/other_context/items[openEHR-EHR-OBSERVATION.pulse_oximetry.v1]/context/start_time",
+            formPath: "vitalSigns.oxygen_saturation.timestamp"
```
Comment (dominikznidar) [inline frontend/local/src/modules/ehr/encounter.js]: the `events` requries the `:0` suffix and the timestamp should be stored as under `/time`

File: frontend/local/src/modules/ehr.js [hunk 10]
```diff
@@ -41,14 +48,41 @@ const buildContextForPatientData = formData => dispatch => {
...
+            // time info
+            "/context/start_time": new Date().toJSON(),
+            "/context/end_time": new Date().toJSON(),
+
+            // // participants
+            // // add doctor
+            "/composer|identifier": author.id,
+            "/composer|name": `${author.personalData.firstName} ${author.personalData.lastName}`,
+
+            "/category": "openehr::433|persistent|"
```
Comment (dominikznidar) [inline frontend/local/src/modules/ehr.js]: please replace `persistent` with `event`

Comment (mateuszkrasucki) [reply frontend/local/src/modules/ehr.js]: done

**Pull request merged**